mtp_cashbook/apps/disbursements/forms.py

```diff
@@ -0,0 +1,183 @@
+                value = getattr(cls, 'serialise_%s' % field)(value)
+            session[field] = value
+
+
+class PrisonerForm(DisbursementForm):
+    prisoner_number = forms.CharField(
+        label='Prisoner number',
+        help_text='For example, A1234BC',
+        max_length=7,
+    )
+    error_messages = {
+        'connection': _('This service is currently unavailable'),
+        'not_found': _('No prisoner matches the details you’ve supplied'),
+        'wrong_prison': _('This prisoner does not appear to be in a prison that you manage'),
+    }
+
+    def clean_prisoner_number(self):
+        prisoner_number = self.cleaned_data.get('prisoner_number')
+        if prisoner_number:
+            prisoner_number = prisoner_number.upper()
+        return prisoner_number
+
+    def clean(self):
+        prisoner_number = self.cleaned_data.get('prisoner_number')
+        session = get_api_session(self.request)
+        try:
+            prisoner = session.get(
+                '/prisoner_locations/{prisoner_number}/'.format(
+                    prisoner_number=prisoner_number
+                )
+            ).json()
+
+            self.cleaned_data['prisoner_name'] = prisoner['prisoner_name']
+            self.cleaned_data['prisoner_dob'] = prisoner['prisoner_dob']
+            self.cleaned_data['prison'] = prisoner['prison']
+            return self.cleaned_data
+        except HttpNotFoundError:
+            raise forms.ValidationError(
+                self.error_messages['not_found'], code='not_found')
+        except Forbidden:
+            raise forms.ValidationError(
+                self.error_messages['wrong_prison'], code='wrong_prison')
+        except RequestException:
+            raise forms.ValidationError(
+                self.error_messages['connection'], code='connection')
+
+
+def serialise_amount(amount):
+    return '{0:.2f}'.format(amount/100)
+
+
+def unserialise_amount(amount_text):
+    amount_text = force_text(amount_text)
+    return Decimal(amount_text)
+
+
+class AmountForm(DisbursementForm):
+    amount = forms.DecimalField(
+        label='Amount to send',
+        help_text='For example, 10.00',
+        min_value=Decimal('0.01'),
+        decimal_places=2,
+    )
+    serialise_amount = serialise_amount
+    unserialise_amount = unserialise_amount
+    error_messages = {
+        'invalid': _('Enter amount as a number'),
+        'min_value': _('Amount should be 1p or more'),
+        'exceeds_funds': _(
+            'There is not enough money in the prisoner’s private account. '
+            'Use NOMIS to move money from other accounts into the private '
+            'account, then click ’Update balances’'),
+    }
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        from .views import PrisonerView
+        try:
+            balances = nomis.get_account_balances(
+                self.previous_form_data[PrisonerView.url_name]['prison'],
```

Comment:
`get_account_balances` does not url-encode so either we should change common or we escape here